hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java

```diff
@@ -631,16 +637,19 @@ private SignatureObject generateSignatureObject(final byte[] hash) {
     private void createSidecarFile(final Builder sidecarFileBuilder, final File sidecarFile)
             throws IOException {
         try (FileOutputStream stream = new FileOutputStream(sidecarFile, false);
+                GZIPOutputStream gzipStream = new GZIPOutputStream(stream);
```

Comment:
This functionality should be behind a feature flag in case it causes a regression and so ops can control its rollout since they currently do the compression in their scripts.